lib/main.dart

```diff
@@ -167,6 +171,14 @@ class _ZapHomePageState extends State<ZapHomePage> with WidgetsBindingObserver {
     }
     return '...';
   }
+  
+  void doesPinExist() async {
+    var pinExists = await Prefs.pinExists();

```

Comment:
remove this function

move this functionality into `_initTokenDetails()`

also call `_initTokenDetails()` on tab change